*** main.m.orig	2013-11-08 15:41:02.000000000 -0600
--- main.m	2013-11-08 15:40:14.000000000 -0600
***************
*** 56,59 ****
--- 56,63 ----
  int launch(char *);
  
+ char **jargv = NULL;
+ int jargc = 0;
+ bool firstTime = true; 
+ 
  int main(int argc, char *argv[]) {
      NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
***************
*** 61,64 ****
--- 65,75 ----
      int result;
      @try {
+         // DRC: I have no idea why this program re-enters itself, but that's
+         // why the "firstTime" check is necessary.
+         if (argc > 1 && !jargv && firstTime) {
+             jargv = &argv[1];
+             jargc = argc - 1;
+         }
+         firstTime = false;
          launch(argv[0]);
          result = 0;
***************
*** 154,158 ****
  
      // Initialize the arguments to JLI_Launch()
!     int argc = 1 + [options count] + 2 + [arguments count] + 1;
      char *argv[argc];
  
--- 165,169 ----
  
      // Initialize the arguments to JLI_Launch()
!     int argc = 1 + [options count] + 2 + [arguments count] + 1 + jargc;
      char *argv[argc];
  
***************
*** 174,177 ****
--- 185,199 ----
      }
  
+     if (jargc > 0 && jargv) {
+         int j;
+         for (j = 0; j < jargc; j++) {
+             if (!strncmp(jargv[j], "-psn", 4)) {
+                 argc--;
+                 continue;
+             }
+             argv[i++] = jargv[j];
+         }
+     }
+ 
      // Invoke JLI_Launch()
      return jli_LaunchFxnPtr(argc, argv,
